### DISCUSSION TITLE
Getting Interpretable Values for Posterior Draws of Parameters with Link Functions

### DISCUSSION THREAD

**Discussion by theonlydvr** [lnccbrown/HSSM#605] [Q&A] [answered]
Most of my model parameters use link functions to maintain bounds but this makes it pretty difficult to interpret values or do custom simulations. I can get average values for the transformed parameters using the Bambi "interpret.predictions" method. Is there an equivalent to get transformed values for each posterior draw instead?

**Comment by digicosmos86** [comment]
Hi @theonlydvr,

HSSM is a wrapper for Bambi, so you can use all bambi utilities. All you need is to access the bambi model through `hssm_model.model`. So for example, you should be able to use `interpret.predictions(hssm_model.model, idata, ...)`.

Please let us know if this works

**Reply by theonlydvr** [reply]
Hi @digicosmos86 

I was able to use interpret.predictions but that only seems to give mean estimates for each parameter at each trial but I was hoping to get corrected estimates for each parameter at every posterior draw. Do you know how I could do that with Bambi/HSSM?

**Reply by digicosmos86** [reply]
Ah sorry for misunderstanding your post. We currently don't have anything that does that in HSSM yet, but looking at `interpret.predictions` in bambi, have you tried specifying the `pps`, `target`, and `transforms` arguments when you are using the function? Although [the documentation](https://bambinos.github.io/bambi/api/interpret.predictions.html#bambi.interpret.predictions) is a bit unclear, it does seem that you can apply transforms when you are calling this function, and it also seems that setting `pps` to `True` might help get estimates for each posterior draw?

**Comment by AlexanderFengler** [comment, marked as answer]
@theonlydvr, 

check out the `add_likelihood_parameters_to_idata()` function. 
This will add the trial wise parameters to the posterior (post link).

You can use those directly for simulation downstream.

**Reply by theonlydvr** [reply]
Perfect, I'll definitely try that out!
